File: pset2/readability/readability.c
/* readability exercise */
//Design and implement a program, readability, that computes the Coleman-Liau index of the text.

//libraries
#include <stdio.h>  // this is needed for printf
#include <cs50.h>   // this is needed for get_string
#include <string.h> // this is needed for strlen
#include <math.h>   // this is needed for round
#include <ctype.h>  // this is needed for isalpha

int main(void)
{
    string str = get_string("Text: ");
    int letters = 0, words = 0, sentences = 0;

    for (int i = 0; i < strlen(str); i++)
    {

        if (isalpha(str[i]))
        {
            letters++;
        }
        else if (str[i] == ' ' || str[i + 1] == '\0')
        {
            words++;
        }

        if (str[i] == '?' || str[i] == '!' || str[i] == '.')
        {
            sentences++;
        }

        // printf("%c - letters: %i; words: %i; sentences: %i\n",str[i], letters, words, sentences);
    }
    // Turn it into floating point division by making at least one operand a floating point number
    float letters_avg = 100 * (float)letters / words;
    float sentences_avg = 100 * (float)sentences / words;
    int cl_idx = round(0.0588 * letters_avg - 0.296 * sentences_avg - 15.8);

    // printf("letters_avg: %f; sentences_avg: %f; cl_idx: %i\n",letters_avg, sentences_avg, cl_idx);

    if (cl_idx < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (cl_idx >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", cl_idx);
    }
}

File: pset6/readability.py
"""
readability exercise: python version
Design and implement a program, readability, that computes the Coleman-Liau index of the text.
Test: https://www.online-utility.org/english/readability_test_and_improve.jsp
"""

# main


def main():
    s = input("Text: ")
    letters = 0
    words = 1  # needs 1 to count the last word
    sentences = 0

# loop string
    for c in s:
        letters += c.isalpha()
        words += c in [' ', '\n']
        sentences += c in ['?', '!', '.']

    # Turn it into floating point division by making at least one operand a floating point number
    letters_avg = 100 * letters / words
    sentences_avg = 100 * sentences / words
    cl_idx = round(0.0588 * letters_avg - 0.296 * sentences_avg - 15.8)

    # print("letters: {}; words: {}; sentences: {}; cl_idx: {}".format(letters, words, sentences, cl_idx))

    if cl_idx < 1:
        print("Before Grade 1")
    elif cl_idx >= 16:
        print("Grade 16+")
    else:
        print("Grade {}".format(cl_idx))


main()
